merino/providers/top_picks.py

```diff
@@ -0,0 +1,160 @@
+"""Top Pick Navigational Queries Provider"""
+import asyncio
+import json
+import logging
+import os
+from collections import defaultdict
+from typing import Any, Optional, Union
+
+from fastapi import FastAPI
+
+from merino.config import settings
+from merino.providers.base import BaseProvider, BaseSuggestion
+
+SCORE: float = settings.providers.top_picks.score
+LOCAL_TOP_PICKS_FILE: str = settings.providers.top_picks.top_picks_file_path
+QUERY_CHAR_LIMIT: int = settings.providers.top_picks.query_char_limit
+
+
+logger = logging.getLogger(__name__)
+
+
+class Suggestion(BaseSuggestion):
+    """Model for Top Pick Query Suggestion"""
+
+    block_id: int
+    domain: str
+    rank: int
+    is_top_pick: bool
+    score: float
+    is_sponsored: bool = False
+
+
+class Provider(BaseProvider):
+    """Top Pick Query Suggestion Provider"""
+
+    # In normal usage this is None, but tests can create the provider with a
+    # FastAPI instance to fetch mock responses from it. See `__init__()`.
+    _app: Optional[FastAPI]
+
+    primary_index: defaultdict = defaultdict(list)
+    secondary_index: defaultdict = defaultdict(list)
+    results: list[dict[str, Union[str, list, int]]]
+    index_char_range = dict[str, int]
+
+    def __init__(
+        self,
+        app: Optional[FastAPI] = None,
+        name: str = "top_picks",
+        enabled_by_default: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        self._app = app
+        self._name = name
+        self._enabled_by_default = enabled_by_default
+        super().__init__(**kwargs)
+
+    async def initialize(self) -> None:
+        """Initialize the provider."""
+        try:
+            index_results_dict: dict = await asyncio.to_thread(Provider.build_indices)
+            self.primary_index: dict = index_results_dict["primary_index"]
+            self.secondary_index: list = index_results_dict["secondary_index"]
+            self.results: list[Suggestion] = index_results_dict["results"]
+            self.index_char_range: dict = index_results_dict["index_char_range"]
+
+        except Exception as e:
+            logger.warning(f"Could not instantiate Top Pick Provider: {e}")
+            raise e
+
+    def hidden(self) -> bool:  # noqa: D102
+        return False
+
+    async def query(self, q: str) -> Any:
```

Comment:
Let's stick to the abstract method defined in providesr/base.py.
```suggestion
    async def query(self, srequest: SuggestionRequest) -> list[BaseSuggestion]:
:
```